org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/DoEntitySerializer.java

```diff
@@ -138,6 +132,29 @@ else if (obj.getClass() == DoEntity.class) {
     }
   }
 
+  /**
+   * Checks if serialization according to declared type should be used. Conditions are:
+   * <ul>
+   * <li>Declared attribute type is available (e.g. attribute is not part of an untyped DoEntity</li>
+   * <li>If lenient mode: Only if attribute type matches the declared attribute type (data object might have an invalid
+   * structure, e.g. string instead of an enum if deserialized lenient)</li>
+   * <li>Attribute type is not a data object type or subclass/subinterface (favor value based serialization for
+   * attributes declared as data object or subclasses/subinterfaces)</li>
+   * </ul>
+   * <br>
+   * NOTE: If this method is changed, check the similar statement in
+   * org.eclipse.scout.rt.jackson.dataobject.DoCollectionSerializer#serializeList(java.lang.Iterable,
+   * com.fasterxml.jackson.core.JsonGenerator, com.fasterxml.jackson.databind.SerializerProvider)
+   *
+   * @return {@code true} if for given {@code declaredAttributeType} and given {@code attributeType} serialization using
+   * the declared type should be used. Returns {@code false} otherwise.
+   */
+  protected boolean isSerializeByDeclaredType(JavaType declaredAttributeType, Class<?> attributeType) {
```

Comment:
Detail: place method below serializeMap (before serializeTypedAttribute) because used by both serializeAttribute/serializeMap and serves as a helper method similar to serializeTypedAttribute

Reply:
Done